src/core/Akka/Util/ByteString.cs

```diff
@@ -646,91 +517,61 @@ public int CopyTo(ref Span<byte> buffer, int index, int count)
         public void WriteTo(Stream stream)
         {
             if (stream == null) throw new ArgumentNullException(nameof(stream));
-
-            foreach (var buffer in _buffers)
-            {
-                stream.Write(buffer.Array, buffer.Offset, buffer.Count);
-            }
+            
+            // TODO: remove this method in future versions of Akka.NET and use System.Memory APIs
+            var array = _memory.ToArray();
+            stream.Write(array, 0, array.Length);
```

Comment:
hmm..

need to sleep on whether chunking could be better.

Note: as mentioned in other ramblings, certain read/write operations on streams and the like with `Memory` may have an intermediate copy to a magical CLR-internal buffer.